docs/glossary.md

```diff
@@ -0,0 +1,204 @@
+# Glossary
+
+## Anonymous branch
+
+An anonymous branch is a chain of commits that doesn't have any
+[named branches](#branch) pointing to it or to any of its descendants. Unlike
+Git, Jujutsu keeps commits on anonymous branches around until they are
+explicitly abandoned.
+
+## Backend
+
+A backend is an implementation of the storage layer. There are currently two
+builtin commit backends: the Git backend and the native backend. The Git backend
+stores commits in a Git repository. The native backend is used for testing
+purposes only. Alternative backends could be used, for example, if somebody
+wanted to use jj with a humongous monorepo (as Google does).
+
+There are also pluggable backends for storing other information than commits,
+such as the "operation store backend" for storing
+[the operation log](#operation-log).
+
+## Branch
+
+A branch is a named pointer to a [commit](#commit). They automatically follow
+the commit if it gets [rewritten](#rewrite). Branches are sometimes called
+"named branches" to distinguish them from
+[anonymous branches](#anonymous-branch), but note that they are more similar
+to Git's branches than to
+[Mercurial's named branches](https://www.mercurial-scm.org/wiki/Branch#Named_branches).
+
+See [here](branches.md) for details.
+
+## Change
+
+A change is a commit as it [evolves over time](#rewrite).
+
+## Change ID
+
+A change ID is a unique identifier for a [change](#change). They are typically
+16 bytes long and are often randomly generated. They are presented as
+hexadecimal numbers at the start of the line in `jj log`, using 12 hexadecimal
+digits by default. However, the hexadecimal digits are taken from the end of the
+alphabet (z-k) instead of the usual 0-9 and a-f.
+
+## Commit
+
+A snapshot of the files in the repository at a given point in time (technically
+a [tree object](#tree)), together with some metadata. The metadata includes the
+author, the date, and pointers to the commit's parents. Through the pointers to
+the parents, the commits form a
+[Directed Acyclic Graph (DAG)](https://en.wikipedia.org/wiki/Directed_acyclic_graph)
+.
+
+Note that even though commits are stored as snapshots, they are often treated
+as differences between snapshots, namely compared to their parent's snapshot. If
+they have more than one parent, then the difference is computed against the
+result of merging the parents. For example, `jj diff` will show the differences
+introduced by a commit compared to its parent(s), and `jj rebase` will apply
+those changes onto another base commit.
+
+The word "revision" is used as a synonym for "commit".
+
+## Commit ID
+
+A commit ID is a unique identifier for a [commit](#commit). They are 20 bytes
+long when using the Git backend. They are presented in regular hexadecimal
```

Comment:
Should we mention that they coincide with Git's commit id when using the Git backend? Or do we not want to promise that?

Reply:
I think it makes sense to mention. If we ever change it, we'll just have to remember to update this text.